External/FEXCore/Source/Interface/IR/Passes/IRDumperPass.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+$info$
+tags: ir|debug
+desc: Prints IR
+$end_info$
+*/
+
+#include "Interface/IR/PassManager.h"
+#include "Interface/IR/Passes/RegisterAllocationPass.h"
+#include "Interface/Core/OpcodeDispatcher.h"
+
+#include <FEXCore/IR/IR.h>
+
+namespace FEXCore::IR::Debug {
+class IRDumper final : public FEXCore::IR::Pass {
+public:
+  IRDumper();
+  bool Run(IREmitter *IREmit) override;
+
+private:
+  FEX_CONFIG_OPT(DumpIR, DUMPIR);
+  bool DumpToFile{};
+  bool DumpToLog{};
+};
+
+IRDumper::IRDumper() {
+  const auto DumpIRStr = DumpIR();
+  if (DumpIRStr == "stderr" || DumpIRStr == "stdout" || DumpIRStr == "no") {
```

Comment:
maybe add a comment in the body like:

```cpp
// Intentionally do nothing
```

just to make it super clear the lack of behavior is intentional

Reply:
comment added.